pkg/logql/log/pipeline_test.go

```diff
@@ -12,23 +12,29 @@ import (
 
 func TestNoopPipeline(t *testing.T) {
 	lbs := labels.FromStrings("foo", "bar")
-	l, lbr, matches := NewNoopPipeline().ForStream(lbs).Process(0, []byte(""))
+	pipeline := NewNoopPipeline().(*noopPipeline)
+
+	l, lbr, matches := pipeline.ForStream(lbs).Process(0, []byte(""))
 	require.Equal(t, []byte(""), l)
 	require.Equal(t, NewLabelsResult(lbs, lbs.Hash()), lbr)
 	require.Equal(t, true, matches)
 
-	ls, lbr, matches := NewNoopPipeline().ForStream(lbs).ProcessString(0, "")
+	ls, lbr, matches := pipeline.ForStream(lbs).ProcessString(0, "")
 	require.Equal(t, "", ls)
 	require.Equal(t, NewLabelsResult(lbs, lbs.Hash()), lbr)
 	require.Equal(t, true, matches)
+
+	pipeline.Reset()
+	require.Len(t, pipeline.cache, 0)
```

Comment:
I wish there was a way to better test the behavior rather than reaching in to private data, but I can't think of one, so I'll just whine instead.